adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -41,6 +41,102 @@ import scala.collection.JavaConversions._
+    val lines = input.split('\n')
+    require(lines.length == 4,
+      s"Input must have 4 lines (${lines.length.toString} found):\n${input}")
+
+    val readName = lines(0).drop(1)
+    if (readName.endsWith("/1") && setSecondOfPair)
+      throw new Exception(
+        s"Found read name $readName ending in '/1' despite second-of-pair flag being set"
+      )
+    else if (readName.endsWith("/2") && setFirstOfPair)
+      throw new Exception(
+        s"Found read name $readName ending in '/2' despite first-of-pair flag being set"
+      )
+    val suffix = """(\/1$)|(\/2$)""".r
+    val readNameNoSuffix = suffix.replaceAllIn(readName, "")
+
+    val readSequence = lines(1)
+    val readQualitiesRaw = lines(3)
+
+    val readQualities =
+      if (stringency == ValidationStringency.STRICT) readQualitiesRaw
+      else {
+        if (readQualitiesRaw == "*") "B" * readSequence.length
+        else if (readQualitiesRaw.length < readSequence.length) readQualitiesRaw + ("B" * (readSequence.length - readQualitiesRaw.length))
+        else if (readQualitiesRaw.length > readSequence.length) throw new NotImplementedError("Not implemented")
+        else readQualitiesRaw
+      }
+
+    if (stringency == ValidationStringency.STRICT) {
+      if (readQualitiesRaw == "*" && readSequence.length > 1)
+        throw new IllegalArgumentException(s"Fastq quality must be defined for\n $input")
+    }
+
+    require(
+      readSequence.length == readQualities.length,
+      s"The first read: ${readName}, has different sequence and qual length."
+    )
+
+    (readNameNoSuffix, readSequence, readQualities)
+  }
+
+  private def parseReadPairInFastq(input: String): (String, String, String, String, String, String) = {
+    val lines = input.toString.split('\n')
+    require(lines.length == 8,
+      s"Record must have 8 lines (${lines.length.toString} found):\n${input}")
+
+    val (firstReadName, firstReadSequence, firstReadQualities) =
+      this.parseReadInFastq(lines.take(4).mkString("\n"), setFirstOfPair = true, setSecondOfPair = false)
+
+    val (secondReadName, secondReadSequence, secondReadQualities) =
+      this.parseReadInFastq(lines.drop(4).mkString("\n"), setFirstOfPair = false, setSecondOfPair = true)
+
+    (
+      firstReadName,
+      firstReadSequence,
+      firstReadQualities,
+      secondReadName,
+      secondReadSequence,
+      secondReadQualities
+    )
+  }
+
+  private def makeAlignmentRecord(readName: String,
+                                  sequence: String,
+                                  qual: String,
+                                  readInFragment: Int,
+                                  readPaired: Boolean = true,
+                                  recordGroupOpt: Option[String] = None): AlignmentRecord = {
+    val builder = AlignmentRecord.newBuilder
+      .setReadName(readName)
+      .setSequence(sequence)
+      .setQual(qual)
+      .setReadPaired(readPaired)
+      .setProperPair(null)
```

Comment:
explicitly setting null is dropped in newer commits. If you rerun the tests, it should all pass.